tests/component/db/bigquery/test_load_config_options_datetime.py

```diff
@@ -0,0 +1,132 @@
+import unittest
+
+from records_mover.db.bigquery.load_job_config_options import load_job_config
+from records_mover.records.load_plan import RecordsLoadPlan
+from records_mover.records.processing_instructions import ProcessingInstructions
+from records_mover.records.records_format import DelimitedRecordsFormat
+from ...records.datetime_cases import (
+    DATE_CASES, DATETIMETZ_CASES, DATETIME_CASES, TIMEONLY_CASES
+)
+
+
+class TestLoadJobConfigDatetime(unittest.TestCase):
+    def test_dateformat(self):
+        processing_instructions = ProcessingInstructions(fail_if_dont_understand=True,
+                                                         fail_if_cant_handle_hint=True,
+                                                         fail_if_row_invalid=True)
+        # Double check this before adding anything else in here to see
+        # if it has changed, but YYYY-MM-DD is the only format
+        # accepted by BigQuery as of this writing
+        should_raise = {
+            'YYYY-MM-DD': False,
+            'MM-DD-YYYY': True,
+            'DD-MM-YYYY': True,
+            'MM/DD/YY': True,
+            'DD/MM/YY': True,
+            'DD-MM-YY': True,
+        }
+        for dateformat in DATE_CASES:
+            records_format =\
+                DelimitedRecordsFormat(variant='bigquery',
+                                       hints={
+                                           'dateformat': dateformat
+                                       })
+            load_plan = RecordsLoadPlan(processing_instructions=processing_instructions,
+                                        records_format=records_format)
+            unhandled_hints = set(records_format.hints.keys())
+            try:
+                load_job_config(unhandled_hints, load_plan)
+            except NotImplementedError:
+                if should_raise[dateformat]:
+                    pass
+                else:
+                    raise
```

Comment:
The general intent with these tests is that as we add new things to the below datetime_cases.py module, tests will blow up and we'll need to go back and hand-add expectations of what each driver should produce for each input.